coderd/database/pubsub/pubsub.go

```diff
@@ -223,12 +229,18 @@ func (p *pgPubsub) subscribeQueue(event string, newQ *msgQueue) (cancel func(),
 		delete(listeners, id)
 
 		if len(listeners) == 0 {
-			_ = p.pgListener.Unlisten(event)
+			uErr := p.pgListener.Unlisten(event)
+			if uErr != nil && !p.closedListener {
+				p.logger.Warn(p.ctx, "failed to unlisten", slog.Error(uErr), slog.F("event", event))
+			} else {
+				p.logger.Debug(p.ctx, "stopped listening to event channel", slog.F("event", event))
+			}
 		}
 	}, nil
 }
 
 func (p *pgPubsub) Publish(event string, message []byte) error {
+	p.logger.Debug(p.ctx, "publish", slog.F("event", event))
```

Comment:
I looked a the failed CI report, and found many entries like this:
```
    t.go:99: 2024-01-31 10:54:08.149 [debu]  publish  event=test
    t.go:99: 2024-01-31 10:54:08.149 [debu]  publish  event=test
    t.go:99: 2024-01-31 10:54:08.149 [debu]  publish  event=test
    t.go:99: 2024-01-31 10:54:08.149 [debu]  publish  event=test
    t.go:99: 2024-01-31 10:54:08.149 [debu]  publish  event=test
    t.go:99: 2024-01-31 10:54:08.150 [debu]  publish  event=test
```

Should we add more context/details describing the event? Maybe first N bytes of the message?

Reply:
I'm nervous about logging the message, as there isn't a good way to ensure things like crypto keys are never sent.

Reply:
Agreed on not logging the data. Alternatively we can make it a practice to never pass data in pubsub but that seems limiting. (E.g. you would pass database row IDs instead of raw data.)